docs/contribute/general/git.rst

```diff
@@ -49,7 +49,26 @@ So this file must at least include the following lines:
 ***************
 
 Before merging a PR all commits shall be squashed into few (desired only
-one) logical commits.
+one) logical commits. This is done following the rules below.
+
+Use ``Squash & Merge`` in case:
+
+   #. there is one initial commit + n "fix findings" commits
```

Comment:
Or any other form of commits that we do not want to preserve: "work in progress" + "blah" + "blub" + "still work in progress"